src/lib.rs

```diff
@@ -23,26 +27,26 @@ fn is_leap_year(year: u32) -> bool {
     (year % 4 == 0) ^ (year % 100 == 0) ^ (year % 400 == 0)
 }
 
-fn days_by_year(mut year: u32) -> u32 {
-    let mut count: u32 = 0;
+fn count_leap_years(year: u32) -> u32 {
+    if year < 1 {
+        0
+    } else if year <= REFORM_YEAR {
```

Comment:
```suggestion
    if year <= REFORM_YEAR {
```

What about remove this first conditional? this already validated in days_by_year function?